### ISSUE TITLE
throws an exception when open and play mp4 file  on windows 

### ISSUE THREAD

**Issue by bewithme** [caprica/vlcj-player#15]
I can open and play mp4 file on MacOSX now, but it throws an exception as  below when on windows
that's why? 

java.lang.ClassCastException: java.awt.image.DataBufferUShort cannot be cast to
java.awt.image.DataBufferInt
        at uk.co.caprica.vlcj.player.component.CallbackMediaPlayerComponent$Defa
ultRenderCallback.setImageBuffer(CallbackMediaPlayerComponent.java:368)
        at uk.co.caprica.vlcj.player.component.CallbackMediaPlayerComponent$Defa
ultRenderCallback.access$600(CallbackMediaPlayerComponent.java:365)
        at uk.co.caprica.vlcj.player.component.CallbackMediaPlayerComponent.newV
ideoBuffer(CallbackMediaPlayerComponent.java:357)
        at uk.co.caprica.vlcj.player.component.CallbackMediaPlayerComponent.acce
ss$500(CallbackMediaPlayerComponent.java:50)
        at uk.co.caprica.vlcj.player.component.CallbackMediaPlayerComponent$Defa
ultBufferFormatCallback.getBufferFormat(CallbackMediaPlayerComponent.java:337)

**Comment by caprica** [comment]
What is your run-time Java version?

**Comment by bewithme** [comment]
> What is your run-time Java version?

jdk 1.8.0_201 

I can play video file or mrl  use VLC media player that I install on windows, but exception throw when I use java edition  vlcj-player

hope you  can fix it ,thank you so much .

**Comment by caprica** [comment]
What code are you running? Is it the unmodified vlcj-player code, or did you change something?

By the way, nothing prevents you using your own RenderCallback rather than the one provided by default.

**Comment by bewithme** [comment]
No,I do nothing to the code,

**Comment by caprica** [comment]
So on Windows, unlike on OSX or Linux, when you do this...
```
image.getRaster().getDataBuffer()).getData();
```
...you apparently get a DataBufferUShort rather than an DataBufferInt.

That makes no sense to me.

**Comment by caprica** [comment]
Would you please try something for me...

In the CallbackMediaPlayerComponent class, find this method (line number 355 onwards):
```
    private void newVideoBuffer(int width, int height) {
        image = GraphicsEnvironment.getLocalGraphicsEnvironment().getDefaultScreenDevice().getDefaultConfiguration().createCompatibleImage(width, height);
        defaultRenderCallback.setImageBuffer(image);
        videoSurfaceComponent.setPreferredSize(new Dimension(width, height));
    }
```
And change the image creation to this instead:
```
    private void newVideoBuffer(int width, int height) {
        image = new BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB);
        defaultRenderCallback.setImageBuffer(image);
        videoSurfaceComponent.setPreferredSize(new Dimension(width, height));
    }
```
And then run your test again.

**Comment by bewithme** [comment]
> Would you please try something for me...
> 
> In the CallbackMediaPlayerComponent class, find this method (line number 355 onwards):
> 
> ```
>     private void newVideoBuffer(int width, int height) {
>         image = GraphicsEnvironment.getLocalGraphicsEnvironment().getDefaultScreenDevice().getDefaultConfiguration().createCompatibleImage(width, height);
>         defaultRenderCallback.setImageBuffer(image);
>         videoSurfaceComponent.setPreferredSize(new Dimension(width, height));
>     }
> ```
> And change the image creation to this instead:
> 
> ```
>     private void newVideoBuffer(int width, int height) {
>         image = new BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB);
>         defaultRenderCallback.setImageBuffer(image);
>         videoSurfaceComponent.setPreferredSize(new Dimension(width, height));
>     }
> ```
> And then run your test again.

haha it is work now ,  thank  you so much !  you are good man, can you tell me how to  draw something on the frame ,  for example bounding box for a car which if I know the position of the car.

**Comment by caprica** [comment]
Is there something unusual or non-standard about your Windows desktop?

Creating a "compatible" image is the cause of your issue here.

And unfortunately, no, I can't tell you how to draw something on the frame because that's way out of the scope of this project. My advise is for you to seek out a Java2D tutorial. If you are using CallbackMediaPlayerComponent you can override onPaintOverlay and do your drawing in there.

**Comment by caprica** [comment]
https://github.com/caprica/vlcj/issues/849

**Comment by bewithme** [comment]
> [caprica/vlcj#849](https://github.com/caprica/vlcj/issues/849)

thank you

**Comment by caprica** [comment]
Fixed in vlcj-4.0.8
